packages/mattermost-redux/src/reducers/entities/threads.test.js

```diff
@@ -38,9 +38,40 @@ describe('threads', () => {
         });
         expect(nextState.threadsInTeam.a).toContain('t1');
     });
+
+    test('RECEIVED_UNREAD_THREADS should update the state', () => {
+        const state = deepFreeze({
+            threadsInTeam: {},
+            unreadThreadsInTeam: {},
+            threads: {},
+            counts: {},
+        });
+
+        const nextState = threadsReducer(state, {
+            type: ThreadTypes.RECEIVED_UNREAD_THREADS,
+            data: {
+                team_id: 'a',
+                threads: [
+                    {id: 't1'},
+                ],
+                total: 3,
+                total_unread_threads: 0,
+                total_unread_mentions: 1,
+            },
+        });
+
+        expect(nextState).not.toBe(state);
+        expect(nextState.threads.t1).toEqual({
+            id: 't1',
+        });
+        expect(nextState.unreadThreadsInTeam.a).toContain('t1');
+        expect(nextState.threadsInTeam).toBe(state.threadsInTeam);
+    });
+
     test('ALL_TEAM_THREADS_READ should clear the counts', () => {
         const state = deepFreeze({
             threadsInTeam: {},
+            unreadThreadsInTeam: {},
```

Comment:
If you want to save some typing here, I realized recently that you can just call the reducer with an empty action object to initialize the empty state. I think that's what redux does internally.

I did that throughout the channels reducer so that I wouldn't have to change every existing test when I added a new field to the state.